### PR TITLE
#8 spring security

### PR DESCRIPTION
**Pull request by yanna-stepanova**

### CODE REVIEW

File: src/main/java/com/yanna/stepanova/model/User.java [hunk 1]
```diff
@@ -32,4 +41,44 @@ public class User {
     private String shippingAddress;
     @Column(name = "is_deleted", nullable = false)
     private boolean isDeleted;
+    @OneToMany(fetch = FetchType.EAGER)
```
Comment (Elena-Bruyako) [inline src/main/java/com/yanna/stepanova/model/User.java]:
```suggestion
    @ManyToMany
```

File: src/main/java/com/yanna/stepanova/service/impl/UserServiceImpl.java [hunk 2]
```diff
@@ -4,23 +4,30 @@
...
 
     @Override
     public UserResponseDto register(UserRegistrationRequestDto requestDto)
             throws RegistrationException {
         if (userRepo.existsByEmail(requestDto.email())) {
             throw new RegistrationException("This user can't be registered");
         }
-        return userMapper.toResponseDto(userRepo.save(userMapper.toModel(requestDto)));
+        User user = userMapper.toModel(requestDto);
+        user.setPassword(passwordEncoder.encode(user.getPassword()));
```
Comment (Elena-Bruyako) [inline src/main/java/com/yanna/stepanova/service/impl/UserServiceImpl.java:36]: Do we need to set up Role.User during the registration or you are doing this in other way?

File: src/main/resources/db/changelog/changes/01-create-table-books.yaml [hunk 3]
```diff
@@ -36,7 +36,7 @@ databaseChangeLog:
                     nullable: false
               - column:
                   name: is_deleted
-                  type: boolean
+                  type: bit
```
Comment (Elena-Bruyako) [inline src/main/resources/db/changelog/changes/01-create-table-books.yaml:39]: Why did you decide to change this type?

File: src/main/resources/db/changelog/changes/04-create-table-users_roles.yaml [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+       id:  create-table-users_roles
+       author:  yana_stepanova
+       changes:
+         - createTable:
+             tableName: users_roles
+             columns:
+               - column:
+                   name: user_id
+                   type: bigint
+                   constraints:
```
Comment (Elena-Bruyako) [inline src/main/resources/db/changelog/changes/04-create-table-users_roles.yaml:12]: foreignKeyName: fk_users_roles_users
referencedTableName: users
referencedColumnNames:  id
nullable:  false

File: src/main/resources/db/changelog/changes/04-create-table-users_roles.yaml [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+                     primaryKey: true
+                     nullable: false
+               - column:
+                   name: role_id
+                   type: bigint
+                   constraints:
```
Comment (Elena-Bruyako) [inline src/main/resources/db/changelog/changes/04-create-table-users_roles.yaml:20]: foreignKeyName: fk_users_roles_roles
referencedTableName: roles
referencedColumnNames:  id
nullable:  false

File: src/main/resources/db/changelog/changes/04_0-add-foreignKeys-users_roles.yaml [hunk 6]
```diff
@@ -0,0 +1,23 @@
+databaseChangeLog:
```
Comment (Elena-Bruyako) [inline src/main/resources/db/changelog/changes/04_0-add-foreignKeys-users_roles.yaml]: you actually don't need this changelog

File: src/main/java/com/yanna/stepanova/security/CustomUserDetailsService.java [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class CustomUserDetailsService implements UserDetailsService {
+    private final UserRepository userRepo;
+
+    @Override
+    public UserDetails loadUserByUsername(String email) throws UsernameNotFoundException {
+        return userRepo.findByEmail(email).orElseThrow(() -> new UsernameNotFoundException(
```
Comment (Elena-Bruyako) [inline src/main/java/com/yanna/stepanova/security/CustomUserDetailsService.java]:
```suggestion
        return userRepo.findByEmail(email).orElseThrow(() -> new EntityNotFoundException(
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
Good job! Let's improve your solution!

File: src/main/resources/db/changelog/changes/01-create-table-books.yaml (hunk 3 above)
Comment (yanna-stepanova) [reply src/main/resources/db/changelog/changes/01-create-table-books.yaml:39]: I had a problem with boolean type when these tables were deleted accidentally. Type 'boolean' started parsing as type 'bit' or 'tinyint'. Even if I used SQL in MySQL for creating table.

Comment (Elena-Bruyako) [reply src/main/resources/db/changelog/changes/01-create-table-books.yaml:39]: Please, check if you didn't, which type preferable 'bit' or 'tinyint' for boolean.

**Review by Elena-Bruyako — APPROVED** [review]
Well done! Check comment.

File: src/main/resources/db/changelog/changes/01-create-table-books.yaml (hunk 3 above)
Comment (yanna-stepanova) [reply src/main/resources/db/changelog/changes/01-create-table-books.yaml:39]: bit: storage size is 1 bit, possible values are 0 or 1
tinyint: storage size is 1 byte, possible values are 0-255

**Pull request merged**